File: app/src/main/java/cedule/app/utils/TimeUtils.java
package cedule.app.utils;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Locale;
import java.util.concurrent.TimeUnit;

public class TimeUtils {
    private static String replenishZero(int target) {
        String targetInString = String.valueOf(target);

        if (2 - targetInString.length() > 0) {
            return "0".repeat(2 - targetInString.length()) + target;
        }
        return targetInString;
    }

    public static String toTimeNotation(int seconds) {
        int hrs = (int) (seconds / 3600.);
        int min = seconds/60 - hrs*60;
        int sec = seconds - hrs * 3600 - min * 60;

        return replenishZero(hrs) + ":" +  replenishZero(min) + ":" + replenishZero(sec);
    }


    public static String toTimeString(long ms) {
        int hour = (int) TimeUnit.MILLISECONDS.toHours(ms);
        int min = (int) (TimeUnit.MILLISECONDS.toMinutes(ms) - hour * 60);

        return replenishZero(hour) + ":" + replenishZero(min);
    }

    public static void setMidNight(Calendar calendar) {
        calendar.set(Calendar.HOUR_OF_DAY, 0);
        calendar.set(Calendar.MINUTE, 0);
        calendar.set(Calendar.SECOND, 0);
        calendar.set(Calendar.MILLISECOND, 0);
    }

    public static String toDateString(long ms) {
        Calendar date = Calendar.getInstance();
        date.setTimeInMillis(ms);

        // For example: 1 -> Feb
        String monthInEng = new SimpleDateFormat("MMM", Locale.ENGLISH).format(ms);

        return date.get(Calendar.DAY_OF_MONTH) + " " + monthInEng + " " + date.get(Calendar.YEAR);
    }
}

File: app/src/main/java/cedule/app/utils/TimeUtils.kt
package cedule.app.utils

import java.text.SimpleDateFormat
import java.util.Calendar
import java.util.Locale
import java.util.concurrent.TimeUnit

object TimeUtils {
    private fun replenishZero(target: Int): String {
        val targetInString = target.toString()

        if (2 - targetInString.length > 0) {
            return "0".repeat(2 - targetInString.length) + target
        }
        return targetInString
    }

    fun toTimeNotation(seconds: Int): String {
        val hrs = (seconds / 3600.0).toInt()
        val min = seconds / 60 - hrs * 60
        val sec = seconds - hrs * 3600 - min * 60

        if (hrs > 0) {
            return replenishZero(hrs) + ":" + replenishZero(min) + ":" + replenishZero(sec)
        }
        return replenishZero(min) + ":" + replenishZero(sec)
    }

    fun toTimeString(ms: Long): String {
        val hour = TimeUnit.MILLISECONDS.toHours(ms).toInt()
        val min = (TimeUnit.MILLISECONDS.toMinutes(ms) - hour * 60).toInt()

        return replenishZero(hour) + ":" + replenishZero(min)
    }

    fun setMidNight(calendar: Calendar): Calendar {
        calendar[Calendar.HOUR_OF_DAY] = 0
        calendar[Calendar.MINUTE] = 0
        calendar[Calendar.SECOND] = 0
        calendar[Calendar.MILLISECOND] = 0

        return calendar
    }

    fun toDateString(ms: Long): String {
        val date = Calendar.getInstance()
        date.timeInMillis = ms

        val month = SimpleDateFormat("MMM", Locale.ENGLISH).format(ms) // e.g. Feb
        return date[Calendar.DAY_OF_MONTH].toString() + " " + month + " " + date[Calendar.YEAR]
    }

    fun getTodayMidnight(): Long {
        val calendar = Calendar.getInstance()
        setMidNight(calendar)

        return calendar.timeInMillis
    }

    fun getAllDatesForYear(year: Int): List<DateInfo> {
        val dateList = mutableListOf<DateInfo>()

        val calendar = setMidNight(Calendar.getInstance())
        calendar.set(year, Calendar.JANUARY, 1)

        val weekdayFormat = SimpleDateFormat("EEE", Locale.ENGLISH) // e.g. Mon
        val dayFormat = SimpleDateFormat("dd", Locale.ENGLISH)      // e.g. 01
        val monthFormat = SimpleDateFormat("MMM", Locale.ENGLISH)   // e.g. Jan

        while (calendar.get(Calendar.YEAR) == year) {
            dateList.add(
                DateInfo(
                    weekday = weekdayFormat.format(calendar.time),
                    day = dayFormat.format(calendar.time).toInt(),
                    month = monthFormat.format(calendar.time),
                    calendar.timeInMillis
                )
            )
            calendar.add(Calendar.DAY_OF_YEAR, 1)
        }

        return dateList
    }
}

data class DateInfo(
    val weekday: String,
    val day: Int,
    val month: String,
    val timestamp: Long
)